esrally/driver/runner.py

```diff
@@ -1535,6 +1539,76 @@ def __repr__(self, *args, **kwargs):
         return "put-settings"
 
 
+class CreateTransform(Runner):
+    """
+    Execute the `create transform API https://www.elastic.co/guide/en/elasticsearch/reference/current/put-transform.html`_.
+    """
+
+    async def __call__(self, es, params):
+        transform_id = mandatory(params, "transform-id", self)
+        body = mandatory(params, "body", self)
+        defer_validation = params.get("defer_validation", False)
+        await es.transform.put_transform(transform_id=transform_id, body=body, defer_validation=defer_validation)
```

Comment:
2 questions (applies to all added runners):

1. according to Runner this should return e.g. 

```
return 1, "ops"
```

However there are also examples of returning complex structures (dicts). What's the right way to do it?

2. if the track contains an error in the config, rally still succeeds and I get no useful information in the logs (only the response code). 

How should error handling be implemented? Shall I `catch/log/raise`?

Reply:
To question 1: It depends whether you want to add additional metadata. See the [docs](https://esrally.readthedocs.io/en/stable/adding_tracks.html#custom-runners) on possible return values. The additional return values are only useful if you configure an Elasticsearch metrics store because the metadata are attached to each metrics sample. If you only use the in-memory metrics store (the implicit default) it's not relevant.

> 2. if the track contains an error in the config, rally still succeeds and I get no useful information in the logs (only the response code).

The reason for this is that Rally continues on any errors by default and only records that an error has happened. If you configure an Elasticsearch metrics store you can see the full error details but they are not recorded in the logs (maybe they should). If you want to be more strict, you can specify `--on-error=abort` when you start the benchmark and Rally will immediately stop the benchmark and print the detailed error message it got from Elasticsearch.

> How should error handling be implemented? Shall I catch/log/raise?

Unless you need vey special handling, like calling the API differently in case the first call fails, there is no need for this. Just let the error bubble up and Rally will handle it as I've described above.